models/userStatus.js

```diff
@@ -0,0 +1,111 @@
+const { userState } = require("../constants/userStatus");
+const firestore = require("../utils/firestore");
+const userStatusModel = firestore.collection("userStatus");
+
+/**
+ * @param userId {string} : id of the user
+ * @returns {Promise<userStatusModel|string>} : returns id of the deleted userStatus
+ */
+const deleteUserStatus = async (userId) => {
+  try {
+    const userStatusDocs = await userStatusModel.where("userId", "==", userId).limit(1).get();
+    const [userStatusDoc] = userStatusDocs.docs;
+    if (userStatusDoc) {
+      const docId = userStatusDoc.id;
+      await userStatusModel.doc(docId).delete();
+      return { id: userStatusDoc.id, userStatusExisted: true, userStatusDeleted: true };
+    }
+  } catch (error) {
+    logger.error(`error in deleting User Status Document . Reason - ${error}`);
+  }
+  return { userStatusExisted: false, userStatusDeleted: false };
+};
+
+/**
+ * @params userId {string} : id of the user
+ * @returns {Promise<userStatusModel|Object>} : returns the userStatus of a single user
+ */
+const getUserStatus = async (userId) => {
+  try {
+    const userStatusDocs = await userStatusModel.where("userId", "==", userId).limit(1).get();
+    const [userStatusDoc] = userStatusDocs.docs;
+    if (userStatusDoc) {
+      const id = userStatusDoc.id;
+      const data = userStatusDoc.data();
+      return { id, userId, ...data, userStatusExists: true };
+    }
+  } catch (error) {
+    logger.error(`error in fetching the User Status Document. Reason - ${error}`);
+  }
+  return { id: undefined, userId, userStatusExists: false };
```

Comment:
can we put it in else

Reply:
As Mahima is blocked on this PR hence will resolve the pending comments in a new PR.[ Issue ticket ](https://github.com/Real-Dev-Squad/website-backend/issues/834) for the same.

Reply:
Since it was not merged, resolved in the same PR.